src/dsymbol/tests.d

```diff
@@ -298,6 +298,76 @@ unittest
 	assert(T2.kind == CompletionKind.variadicTmpParam);
 }
 
+unittest
+{
+	writeln("Running public import tests...");
+
+	const dir = buildPath(tempDir(), "dsymbol");
+	const fnameA = buildPath(dir, "a.d");
+	const fnameB = buildPath(dir, "b.d");
+	const fnameC = buildPath(dir, "c.d");
+	const fnameD = buildPath(dir, "d.d");
+	const srcA = q{ int x; int w; };
+	const srcB = q{ float y; private float z; };
+	const srcC = q{ public { import a : x; import b; } import a : w; long t; };
+	const srcD = q{ public import c; };
```

Comment:
I think for people glancing over the source code it would be nice to include a small graph describing these relations here:
```d
	//      [  a  ] (x, w)
	//         | public import x
	//         | import w
	//         v
	//      [  c  ] (t + x, y, private w)  -->  [  d  ] (+ t, x, y)
	//         ^
	//         | public import
	//         |
	//      [  b  ] (y, private z)
	//
	// Legend:
	// [ module ] (defined, symbols + imported, symbols)
```
just as a thought though

Reply:
Maybe something simpler:
```
	// A simple diagram:
	// a = x w
	// b = y [z]
	// c = t + (x y) [w]
	// d = (t x y)
```